src/Native/Runtime/coreclr/gcinfodecoder.cpp

```diff
@@ -1476,23 +1476,27 @@ OBJECTREF* GcInfoDecoder::GetRegisterSlot(
     _ASSERTE(regNum >= 0 && regNum <= 14);
     _ASSERTE(regNum != 13);  // sp
 
+#ifdef FEATURE_REDHAWK
+    PTR_UIntNative *ppReg;
+#else
     DWORD **ppReg;
+#endif
 
     if(regNum <= 3)
     {
-        ppReg = &pRD->volatileCurrContextPointers.R0;
+        ppReg = &pRD->pR0;
         return (OBJECTREF*)*(ppReg + regNum);
     }
```

Comment:
Once you do that, the `FEATURE_REDHAWK` path can be simplified. It can be just:

```
    PTR_UIntNative* ppReg = &pRD->pR0;
    if (regNum > 12) regNum--; // rsp is skipped in Redhawk RegDisplay
     return (OBJECTREF*)*(ppReg + regNum);
```